settings.gradle.kts

```diff
@@ -351,6 +352,32 @@ gradle.rootProject {
         outputFile = layout.projectDirectory.file("architecture/platforms.md")
         elements = provider { architectureElements.map { it.build() } }
     }
+    tasks.register("platformsData", GeneratePlatformsDataTask::class) {
+        description = "Generates the platforms data"
+        outputFile = layout.projectDirectory.file("build/architecture/platforms.json")
+        platforms = provider { architectureElements.filterIsInstance<PlatformBuilder>().map { it.build() } }
+    }
+}
+
+abstract class GeneratePlatformsDataTask : DefaultTask() {
+
+    @get:OutputFile
+    abstract val outputFile: RegularFileProperty
+
+    @get:Input
+    abstract val platforms: ListProperty<Platform>
+
+    @TaskAction
+    fun action() {
+        val allPlatforms = platforms.get()
+        // name: [dirs: ..., uses: ...]
```

Comment:
Same comment as before, maybe add a half sentence of comments here and there to specify what "dirs" and "uses" are.

Reply:
Using an explicit class for serialization makes it more obvious